src/plugins/tcl/tcl.cpp

```diff
@@ -42,15 +42,17 @@ int elektraTclGet (Plugin *, KeySet * returned, Key * parentKey)
 #include "readme_tcl.c"
 				     keyNew ("system/elektra/modules/tcl/infos/version", KEY_VALUE, PLUGINVERSION, KEY_END), KS_END));
 		ksDel (n);
+		return ELEKTRA_PLUGIN_STATUS_SUCCESS;
 	}
 	/* get all keys */
 
 	int errnosave = errno;
-	std::ifstream in (keyString (parentKey), std::ios::binary); // we get our input from this file
+	std::ifstream in (parent.getString (), std::ios::binary); // we get our input from this file
 	if (!in.is_open ())
 	{
-		ELEKTRA_SET_ERROR_GET (parentKey);
+		ELEKTRA_SET_ERROR_GET (*parent);
 		errno = errnosave;
+		parent.release ();
```

Comment:
If you happen to start a new C++ plugin: the type plugin solves this much more nicely with a delegate. Then you only have single place where you have to release the ownership.

Reply:
> If you happen to start a new C++ plugin: the type plugin solves this much more nicely with a delegate.

I know, I used the delegate approach for the [CCode plugin updates](https://github.com/ElektraInitiative/libelektra/pull/2066/files#diff-8a51e9df9c5ae96399489342adaa7158R21) (that are part of this PR).